### PR TITLE
Add Tx String representation

### PR DESCRIPTION
**Pull request by jaekwon**
What are the other Objects whose String reprs we can improve? @melekes

### CODE REVIEW

**Comment by adrianbrink** [comment]
@jaekwon @melekes Can this be merged or are we planning to add more things to it?

**Comment by melekes** [comment]
Yes! I did a couple of improvements in my logger PR (already merged)

File: types/tx.go [hunk 1]
```diff
@@ -18,6 +19,10 @@ func (tx Tx) Hash() []byte {
 	return merkle.SimpleHashFromBinary(tx)
 }
 
+func (tx Tx) String() string {
+	return fmt.Sprintf("Tx{%X}")
```
Comment (ebuchman) [inline types/tx.go]: This should not have been merged. It's not correct :)

Comment (melekes) [inline types/tx.go]: Why? Because of the format?

Comment (ebuchman) [inline types/tx.go]: it's missing the argument! should be `("Tx{%X}", tx)`

Comment (melekes) [inline types/tx.go]: oops

Comment (ebuchman) [inline types/tx.go]: no worries. fixed on develop

**Pull request merged**